### PR TITLE
Custom attribute to force sql data type in auto schema generation

### PR DESCRIPTION
**Pull request by CactusChaos**
Added custom data-type attribute to force data type in auto schema generation. Currently I have only added `TEXT` type but others could easily be added into the `sql-type-map`

### CODE REVIEW

File: src/main/com/fulcrologic/rad/database_adapters/sql/migration.clj [hunk 1]
```diff
@@ -30,18 +30,24 @@
    :symbol   "VARCHAR(200)"
    :uuid     "UUID"})
 
+;; Add any additional data types you require here
+(def sql-type-map
+  {:text "TEXT"})
```
Comment (awkay) [inline src/main/com/fulcrologic/rad/database_adapters/sql/migration.clj]: I don't think we need a map. Let's make an option that allows them to just give the literal string of the data type they want. Period.

File: src/main/com/fulcrologic/rad/database_adapters/sql_options.cljc [hunk 2]
```diff
@@ -95,4 +95,8 @@
    isComponent flag.  This is different than SQL CASCADE, because in this case nothing
    has been deleted. The reference on which this option lives is just being changed
    to point to a different thing."
-  :com.fulcrologic.rad.database-adapters.sql/delete-referent?)
\ No newline at end of file
+  :com.fulcrologic.rad.database-adapters.sql/delete-referent?)
+
+(def data-type
+  "Attribute option. Force a data type for the desired attribute when auto generating the database."
```
Comment (awkay) [inline src/main/com/fulcrologic/rad/database_adapters/sql_options.cljc]: Change this to be a string, which is a database-specific data type that will be taken literally as an override.

File: src/main/com/fulcrologic/rad/database_adapters/sql/migration.clj (hunk 1 above)
Comment (CactusChaos) [reply src/main/com/fulcrologic/rad/database_adapters/sql/migration.clj]: Yeah that makes way more sense! Done

**Comment by CactusChaos** [comment]
Changed to string to specify attribute

File: src/main/com/fulcrologic/rad/database_adapters/sql/migration.clj [hunk 3]
```diff
@@ -31,17 +31,19 @@
...
-      (str "VARCHAR(" max-length ")")
-      "VARCHAR(200)")
-    (if-let [result (get type-map type)]
-      result
-      (do
-        (log/error "Unsupported type" type)
-        "TEXT"))))
+                  ::rad.sql/keys [data-type max-length]}]
+       [::attr/attribute => string?]
+       (if-let [result data-type]
```
Comment (awkay) [inline src/main/com/fulcrologic/rad/database_adapters/sql/migration.clj:36]: Seems like `(if data-type data-type ...)` would be better.

**Review by awkay — APPROVED** [review]
Good enough. Thanks!

**Pull request merged**